modules/transport-netty4/src/main/java/org/elasticsearch/http/netty4/Netty4HttpRequestBodyStream.java

```diff
@@ -81,18 +86,23 @@ public void next() {
                 channel.read();
             } else {
                 try {
+                    activityTracker.startActivity();
                     send();
-                } catch (Exception e) {
-                    channel.pipeline().fireExceptionCaught(e);
+                } catch (Throwable t) {
```

Comment:
die-with-dignity test throws OOM exception which is discarded by evenloop executor, need to catch and throw into pipeline